server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -797,8 +797,8 @@ public IntervalsSource getSource(QueryShardContext context, MappedFieldType fiel
             BytesRef normalizedTerm = analyzer.normalize(fieldType.name(), term);
             FuzzyQuery fq = new FuzzyQuery(new Term(fieldType.name(), normalizedTerm),
                 fuzziness.asDistance(term), prefixLength, 128, transpositions);
-            CompiledAutomaton ca = new CompiledAutomaton(fq.toAutomaton());
-            source = XIntervals.multiterm(ca, term);
+            CompiledAutomaton[] automata = fq.getAutomata();
+            source = XIntervals.multiterm(automata[automata.length - 1], term);
             if (useField != null) {
```

Comment:
I think we can remove `XIntervals` now that LUCENE-9050 is released - just use plain `Intervals` instead.

Reply:
@romseygeek Thanks for the comment. Looks like `Intervals.multiterm` expects `Automaton`, while a `FuzzyQuery` can only produce `CompiledAutomaton`.  Or is there any other way to use `Intervals.multiterm`?

Reply:
Hm, maybe we need to put a `fuzzy` method directly on `Intervals` then.  Let's leave `XIntervals` in there for the moment, I'll open a lucene ticket to clean this up.